src/Client.php

```
@@ -443,6 +450,23 @@ public function call($method, $headers = [], $body = '')
                     return $possibleHost !== $hostName;
                 }, ARRAY_FILTER_USE_KEY);
             }
+
+            // All non blacklisted hosts failed, this could be because we are in the middle of a cluster version flip.
+            // ie: We have version 1 and version 2, we've installed version 2 in less than half the cluster, a previous
+            // request already marked all servers in version 2 as failed (since 1 is the current version) in this request
+            // we have installed version 2 in one more server, making it the current version. So now all the servers that
+            // were marked as failed in the previous request are the ones serving requests and all the ones that were good
+            // before are now in the old version and not serving requests. So to cover this, we retry in all servers
+            // once hoping it will find a server that works.
+            if ($exception) {
+                if (!$retriedAllHosts) {
```

NABish you can make this a fail fast:
```
if $exception
   if $retriedAllHosts
       throw
   ...
```